zerver/lib/request.py

```diff
@@ -456,7 +456,30 @@ def _wrapped_req_func(
 
             kwargs[func_var_name] = val
 
-        return req_func(request, *args, **kwargs)
+        return_value = req_func(request, *args, **kwargs)
+
+        if (
+            isinstance(return_value, MutableJsonResponse)
+            and not request_notes.is_webhook_view
+            and 200 <= return_value.status_code < 300
```

Comment:
What's the thinking around this being only for 20x responses? I would expect this feature to be very helpful for some 40x responses in theory (e.g. where you had a typo in a parameter name and are trying to debug). But maybe that makes sense to try to change in a follow-up (possibly not an immediate one), since I can imagine cases where the exact structure through which exceptions are processed could lead to inaccurate output.

Maybe worth a bit of a comment here explaining that consideration.